### PR TITLE
doc(all): move documentation to CLI repo

### PR DESCRIPTION
**Pull request by JeroenVinke**
closes https://github.com/aurelia/cli/issues/719
closes https://github.com/aurelia/cli/issues/709
closes https://github.com/aurelia/cli/issues/699

### CODE REVIEW

File: doc/article/en-US/basics.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## [Running Your Aurelia App](aurelia-doc://section/4/version/1.0.0)
+
+From inside your project folder, simply execute `au run`. This will build your app, creating all bundles in the process. It will start a minimal web server and serve your application. If you would like to develop, with auto-refresh of the browser, simply specify the `--watch` flag like this: `au run --watch`. If you have chosen to use ASP.NET Core and Webpack, you will want to use the `dotnet run` command instead of `au run`.
+
+## [Environments](aurelia-doc://section/5/version/1.0.0)
+
+The CLI build system understands that you might run your code in different environments. By default, you are set up with three: `dev`, `stage` and `prod`. You can use the `--env` flag to specify what environment you want to run under. For example: `au run --env prod --watch`.
+
+When you do a build the `src/environment.js` or `src/environment.ts` file will be overwritten by either the `dev.js`, `stage.js` or `prod.js` file from the `aurelia_project/environments` directory.
```
Comment (JeroenVinke) [inline doc/article/en-US/basics.md]: this is new

File: doc/article/en-US/basics.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Aurelia CLI apps always run in bundled mode, even during development. To build your app, simply run `au build`. You can also specify an environment to build for. For example: `au build --env stage`.
+
+## [Generators](aurelia-doc://section/7/version/1.0.0)
+
+Executing `au generate <resource>` runs a generator to scaffold out typical Aurelia constructs. Options for *resource* are: element, attribute, value-converter, binding-behavior, task and generator. That's right...there's a generator generator so you can write your own. Ex. `au generate element`
+
+## [Aurelia.json](aurelia-doc://section/8/version/1.0.0)
+In the `aurelia_project` directory there is a file called `aurelia.json`. This file is a centralized file containing all settings you might need for your gulp tasks and configuration files. Since it's in a JSON format, it's easy to import from Node.js. The aurelia.json file is meant to be used by your tooling, not your application. 
+
+## [Webpack vs SystemJS vs RequireJS](aurelia-doc://section/9/version/1.0.0)
```
Comment (JeroenVinke) [inline doc/article/en-US/basics.md]: this is new

File: doc/article/en-US/gulp-tasks.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
```
Comment (JeroenVinke) [inline doc/article/en-US/gulp-tasks.md]: this entire document is new

File: doc/article/en-US/migrating.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (JeroenVinke) [inline doc/article/en-US/migrating.md]: this entire document is new

File: doc/article/en-US/updating.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+---
```
Comment (JeroenVinke) [inline doc/article/en-US/updating.md]: this entire document is new

File: doc/article/en-US/webpack.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
```
Comment (JeroenVinke) [inline doc/article/en-US/webpack.md]: this entire document is new

File: doc/article/en-US/aurelia-cli-bundler.md [hunk 7]
```diff
@@ -0,0 +1,570 @@
...
+  	"url": "http://jeroenvinke.nl"
+  },
+  "contributors": [],
+  "translators": [],
+  "keywords": ["CLI", "Bundler"]
+}
+---
+
+## [Introduction](aurelia-doc://section/1/version/1.0.0)
+This part of the documentation is only relevant if you are using the Aurelia CLI Bundler. If you're using Webpack then the Webpack chapter is what you need.
```
Comment (JeroenVinke) [inline doc/article/en-US/aurelia-cli-bundler.md]: this is new

File: doc/article/en-US/aurelia-cli-bundler.md [hunk 8]
```diff
@@ -0,0 +1,570 @@
...
+      "stage & prod": {
+        "max-line-len": 100000
+      } 
+    },
+  </source-code>
+</code-listing>
+
+The Aurelia-CLI uses [UglifyJS2](https://github.com/mishoo/UglifyJS2) for minification, so any option that UglifyJS2 supports is also supported by the Aurelia-CLI. With the above configuration, minification will occur for the `stage` and `prod` environments, but not for the `dev` environment. For the `stage` and `prod` environments, both the `indent_level` as well as the `max-line-len` option are passed to the minifier. The `default` key is optional, but allows you to reduce code duplication when multiple environments have similar options.
+
+## [Loader plugins](aurelia-doc://section/13/version/1.0.0)
```
Comment (JeroenVinke) [inline doc/article/en-US/aurelia-cli-bundler.md]: this is new

File: doc/article/en-US/aurelia-cli-bundler.md [hunk 9]
```diff
@@ -0,0 +1,570 @@
...
+    "elements": "resources/elements",
+    "attributes": "resources/attributes",
+    "valueConverters": "resources/value-converters",
+    "bindingBehaviors": "resources/binding-behaviors",
+    "json": "../node_modules/requirejs-json/json"
+  },
+  </source-code>
+</code-listing>
+
+## [Path mappings](aurelia-doc://section/14/version/1.0.0)
```
Comment (JeroenVinke) [inline doc/article/en-US/aurelia-cli-bundler.md]: this is new

**Comment by JeroenVinke** [comment]
@zewa666 if you have time, could you take a look?

**Comment by ghiscoding** [comment]
@JeroenVinke, correct me if I'm wrong but when I tried the new CLI and WebPack, I had the impression that the `au run` was not working and I had to still use `npm start` for WebPack to work. If `au run` works, then is it missing from the official release `0.31.3`? The reason I brought this up is because there is no mention of that in your updated doc.

**Comment by JeroenVinke** [comment]
@ghiscoding `au run` should work. What problem are you having with the command?

**Comment by ghiscoding** [comment]
It wasn't working at all when I tried which is why I thought `npm start` was still the way to go. I'll have to retry it then, probably be in the weekend after working hours. Thanks

**Comment by EisenbergEffect** [comment]
@JeroenVinke Can you update the format of the articles to match the new format for the docs? I switched the header to a much simplified Yaml format and removed the strange links in the h2s. You can look at any other document in the framework repo for an example. Thanks!

**Comment by JeroenVinke** [comment]
@EisenbergEffect done

I think we can merge this. I've used the below config for the site-generator. If I can create a PR for this somewhere let me know

```
"items": [{
              "name": "Basics",
              "src": "../cli/doc/article/en-US/basics.md",
              "dest": "docs/cli/basics",
              "featured": true
            },
            {
              "name": "Gulp tasks",
              "src": "../cli/doc/article/en-US/gulp-tasks.md",
              "dest": "docs/cli/gulp-tasks"
            },
            {
              "name": "Aurelia-CLI Bundler",
              "src": "../cli/doc/article/en-US/aurelia-cli-bundler.md",
              "dest": "docs/cli/aurelia-cli-bundler"
            },
            {
              "name": "Webpack",
              "src": "../cli/doc/article/en-US/webpack.md",
              "dest": "docs/cli/webpack"
            },
            {
              "name": "Migrating",
              "src": "../cli/doc/article/en-US/migrating.md",
              "dest": "docs/cli/migrating"
            },
            {
              "name": "Updating",
              "src": "../cli/doc/article/en-US/updating.md",
              "dest": "docs/cli/updating"
            },
            {
              "name": "Recipes & known issues",
              "src": "../cli/doc/article/en-US/recipes.md",
              "dest": "docs/cli/recipes"
            }
          ]
```

**Comment by EisenbergEffect** [comment]
Very cool!

**Comment by EisenbergEffect** [comment]
@JeroenVinke I'll take the config you have above and add it into the site config. We'll get this out shortly.

**Pull request merged**